monai/networks/nets/flexible_unet.py

```diff
@@ -243,26 +249,26 @@ def __init__(
         """
         super().__init__()
 
-        if backbone not in encoder_feature_channel:
-            raise ValueError(f"invalid model_name {backbone} found, must be one of {encoder_feature_channel.keys()}.")
+        if backbone not in BACKBONE.register_dict:
+            raise ValueError(f"invalid model_name {backbone} found, must be one of {BACKBONE.register_dict.keys()}.")
 
         if spatial_dims not in (2, 3):
             raise ValueError("spatial_dims can only be 2 or 3.")
 
-        adv_prop = "ap" in backbone
-
+        encoder = BACKBONE.register_dict[backbone]
         self.backbone = backbone
         self.spatial_dims = spatial_dims
-        model_name = backbone
-        encoder_channels = _get_encoder_channels_by_backbone(backbone, in_channels)
-        self.encoder = EfficientNetBNFeatures(
-            model_name=model_name,
-            pretrained=pretrained,
-            in_channels=in_channels,
-            spatial_dims=spatial_dims,
-            norm=norm,
-            adv_prop=adv_prop,
+        encoder_parameters = encoder["parameter"]
+        assert (
```

Comment:
`assert` is not recommended to use in python, please use below logic instead of `assert`:
```py
if "spatial_dims" not in encoder_parameters or "in_channels" not in encoder_parameters or "pretrained" not in encoder_parameters:
    raise ValueError("XXXX")
```